### PR TITLE
Documentation Improvements [v2]

### PR DESCRIPTION
**Pull request by apahim**
v2:
 - Remove parenthesis from release names.
 - Reorganize the commits.

v1: #1317 
- Add captions to separate sections in readthedocs left menu.
- Remove the heads as we are now using captions.
- Fix anchors.
- Create a release_notes directory and an dedicated index file.
- Another batch of release notes.

### CODE REVIEW

**Comment by apahim** [comment]
@ldoktor I quickly tried to change the release notes index for a non-bullet & aligned version, but my first impression it that it's kinda hackish. Let's keep as is for now as it can be improved later.

**Comment by apahim** [comment]
@ldoktor @clebergnu right now readthedocs is facing an outage in building process, so you cannot see this version result. The previous version (http://apahim-avocado.readthedocs.io/en/improve_docs/release_notes/index.html) is still available anyway and the only visible change in v2 is the parenthesis removal.

**Comment by ldoktor** [comment]
No problem, I'm using `make html` anyway :wink:

**Comment by ldoktor** [comment]
Very nice, it's actually quite aligned and looks nice (at least locally using `make html`, we'll see what it will look like on readthedocs). Do you plan to add the rest of the release notes? Or at least the names?

File: docs/source/release_notes/0_25_0.rst [hunk 1]
```diff
@@ -0,0 +1,148 @@
+======
+0.25.0
```
Comment (ldoktor) [inline docs/source/release_notes/0_25_0.rst]: Blade

File: docs/source/release_notes/0_26_0.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
+======
+0.26.0
```
Comment (ldoktor) [inline docs/source/release_notes/0_26_0.rst]: The Office

File: docs/source/release_notes/0_27_0.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
+======
+0.27.0
```
Comment (ldoktor) [inline docs/source/release_notes/0_27_0.rst]: Terminator: Genisys

**Comment by apahim** [comment]
I could not find the names in the release notes for those. I will include now. Thank you!

**Comment by apahim** [comment]
@ldoktor last commit updated.

**Comment by ldoktor** [comment]
OK, thanks, let's get this in and we can add the older release notes later...

**Pull request merged**